pkg/operator/watchdog/cmd.go

```diff
@@ -0,0 +1,247 @@
+package watchdog
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/config/client"
+	"github.com/openshift/library-go/pkg/controller/fileobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/retry"
+)
+
+type FileWatcherOptions struct {
+	ProcessName string
+	Files       []string
+	KubeConfig  string
+	Namespace   string
+	Interval    time.Duration
+
+	recorder       events.Recorder
+	initialContent map[string][]byte
+
+	// for unit-test
+	getPIDByNameFn       func(name string) (int, bool, error)
+	signalFn             func(pid int) error
```

Comment:
what is this? godoc

Reply:
thought the `for unit-test` applies to all below ;-)

Reply:
i will add them

Reply:
it's more like `handleFileChangeFn(pid int)`, right?